src/Zicht/Bundle/UrlBundle/Service/UrlValidator.php

```diff
@@ -18,55 +18,50 @@
 class UrlValidator
 {
     /**
-     * @var Client
-     */
-    private $client;
-
-    private $invalidStatusCodes;
-
-    /**
-     * UrlValidator constructor.
+     * Returns true when url does not return error codes or not found.
      *
-     * @param Client $client
+     * @param string $url
+     * @return boolean
      */
-    public function __construct(Client $client)
+    public function validate($url)
     {
-        $this->client = $client;
-        $this->invalidStatusCodes = array(
-            400,
-            401,
-            403,
-            404,
-            405,
-            500,
-            501,
-            502,
-            503,
-            504,
-            511,
-        );
+        if (null !== ($headers = $this->getHeader($url))) {
+            $statusCode = $this->getStatusCode($headers);
+            // see https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html
+            return ($statusCode >= 200 && $statusCode < 300);
+        }
+        return false;
     }
 
     /**
-     * Returns true when url does not return error codes or not found.
+     * Fetch a url with a HEAD request because we just want to check status code.
      *
      * @param string $url
-     * @return boolean
+     * @return string|bool
      */
-    public function validate($url)
+    protected function getHeader($url)
     {
-        try {
-            $response = $this->client->get($url);
-        } catch (ConnectException $e) {
-            return false;
-        } catch (RequestException $e) {
-            return false;
+        if (false !== @file_get_contents($url, false, stream_context_create(['http' => ['method' => 'HEAD']]))) {
```

Comment:
een @ is een hele dure operatie in PHP. en er zijn nog heel veel andere bezwaren hiertegen. het is een slechte en ongewenste vorm van error handling. http://php.net/manual/en/language.operators.errorcontrol.php lees het ff na zou ik zeggen. en kijk http://php.net/manual/en/function.file-get-contents.php hier wat voor errors dit geeft.

Reply:
ok, weet niet waar je dat heb gelezen en net even getest maar dat is echt te verwaarlozen. 

zonder @
```
real	0m0.150s
```
met @
```
real	0m0.239s
```

over een iteratie van 1000000

Misschien handig ook even in te lezen wat de functie doet. Ik wil hier alleen head reqeust doen en als die een warning geeft boeit hetmij niet want dat zie ik als url die ik niet kan bereiken. Daarom heb ik de @ ervoor gezet (zodat de applicatie geen log geeft of zelfs stopt).

Zolang we niet met 7 zitten is dit de enigste mogelijkheid om dit tegen te gaan. Dus tenzij je ergen not wt anders heb gelezen lijkt met voor dit een gewoon goed oplossing.

Reply:
je kiest hier voor de gebruiker van de library hoe de error handling word gedaan. dit is een url-bundle het is geen error-handler.

Reply:
Misschien beter als niet verder op in gaan en zo laten?

Reply:
Zou het ook zo laten, een `try catch` zou mooier zijn maar dat kan pas in versie 7, dus helaas `@` is niet mooi, maar voor hier en nu de lesser evil.